mobile/lib/components/ramowka.dart

```diff
@@ -0,0 +1,398 @@
+import 'dart:convert';
+
+import 'package:collection/collection.dart';
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+import 'package:intl/intl.dart';
+import 'package:radioaktywne/components/color_shadowed_card.dart';
+import 'package:radioaktywne/components/ra_list_widget.dart';
+import 'package:radioaktywne/extensions/extensions.dart';
+
+/// Widget representing Ramówka
+///
+/// Consists of a [ColorShadowedCard] with a header and
+/// a list of entries in Ramowka.
+class RamowkaWidget extends StatelessWidget {
+  const RamowkaWidget({
+    super.key,
+    this.dataSource,
+    this.timeout = const Duration(seconds: 7),
+  });
+
+  /// Asynchronous source of data in the widget.
+  ///
+  /// On default, the data is pulled from the radioaktywne.pl api.
+  /// If specified, the data will be pulled from the specified source.
+  final Future<void>? dataSource;
+
+  /// Timeout for the fetching function.
+  final Duration timeout;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(
+        vertical: 8,
+        horizontal: 16,
+      ),
+      child: ColorShadowedCard(
+        shadowColor: context.colors.highlightBlue,
+        header: Padding(
+          padding: const EdgeInsets.only(left: 3),
+          child: SizedBox(
+            height: 31,
+            child: Material(
+              color: Colors.transparent,
+              child: InkWell(
+                // TODO: (optional) nagivation to Ramowka page
+                onTap: () {},
+                highlightColor: Colors.transparent,
+                splashColor: context.colors.highlightGreen.withOpacity(0.3),
+                radius: 100,
+                child: Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                  children: [
+                    GestureDetector(
+                      // TODO: nagivation to Ramowka page
+                      onTap: () {
+                        // print("Tapped Ramowka");
+                      },
+                      child: Text(
+                        'Ramówka',
+                        style: context.textStyles.textMedium,
+                      ),
+                    ),
+                    GestureDetector(
+                      // TODO: nagivation to Ramowka page
+                      onTap: () {
+                        // print("Clicked button");
+                      },
+                      child: Icon(
+                        Icons.menu,
+                        size: 28,
+                        color: context.colors.highlightGreen,
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+            ),
+          ),
+        ),
+        child: RamowkaList(
+          timeout: timeout,
+          dataSource: dataSource,
+        ),
+      ),
+    );
+  }
+}
+
+/// List of Ramowka entries.
+class RamowkaList extends StatefulWidget {
```

Comment:
why not [hooks](https://pub.dev/packages/flutter_hooks)?

Reply:
I think it's the rare case that doing it with StatefulWidget is easier because of how FutureBuilder widget works (it needs a future that is instantiated once - at the creation of the widget).